Civi/Api4/Service/Spec/Provider/BatchCreationSpecProvider.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+
+namespace Civi\Api4\Service\Spec\Provider;
+
+use Civi\Api4\Service\Spec\FieldSpec;
+use Civi\Api4\Service\Spec\RequestSpec;
+
+class BatchCreationSpecProvider implements Generic\SpecProviderInterface {
+
+  /**
+   * @inheritDoc
+   */
+  public function modifySpec(RequestSpec $spec) {
+    $spec->getFieldByName('created_id')->setDefaultValue('user_contact_id');
+    $spec->getFieldByName('created_date')->setDefaultValue('now');
+    $spec->getFieldByName('modified_id')->setDefaultValue('user_contact_id');
+    $spec->getFieldByName('modified_date')->setDefaultValue('now');
+    $spec->getFieldByName('status_id')->setRequired(TRUE);
```

Comment:
is there a reason why we aren't setting these to be required at the DB level?

Reply:
Yeah I found that it is set true in DB level as per schema [here](https://github.com/civicrm/civicrm-core/blob/master/xml/schema/Batch/Batch.xml#L141) but in addition I think the API4 spec still need a declaration to indicate the field is required for validation and on API explorer UI as there are other such occurrences in core
```
civicrm-core$ grep -irn "setRequired(TRUE" Civi/Api4/Service/Spec/ | wc -l
      23
```

Reply:
I would have thought that its only needed if the db isn't requiring it right @colemanw ?

Reply:
My bad, yes you are right. On further check it seems like the spec wrapper add the required property as per DAO::fields information. Updated the PR and removed the setRequired on status_id and title. Although I will submit a separate PR to remove the `setRequired` call on required (not pseudo) fields in other EntitySpecProvider files. Thanks for raising this.